File: ui/src/app/business/cluster/cluster-detail/cluster-grade/grade.ts
import {BaseModel} from '../../../../shared/class/BaseModel';

export class Grade extends BaseModel {
    grade: string;
    score: number;
    totalSum: Summary;
    listSum: Summary[] = [];
    results: NamespaceResult[] = [];
}

export class Summary {
    success: number;
    warning: number;
    danger: number;
}

export class NamespaceResult {
    namespace: string;
    results: NamespaceResultDetail[] = [];
}

export class NamespaceResultDetail {
    kind: string;
    name: string;
    podResults: PodResult[] = [];
}

export class PodResult {
    category: string;
    id: string;
    message: string;
    severity: string;
    success: boolean;
}

File: pkg/dto/grade.go
package dto

type ClusterGrade struct {
	Score    int                `json:"score"`
	TotalSum Summary            `json:"totalSum"`
	ListSum  map[string]Summary `json:"listSum"`
	Results  []NamespaceResult  `json:"results"`
}

type Summary struct {
	Danger  int `json:"danger"`
	Warning int `json:"warning"`
	Success int `json:"success"`
}

type NamespaceResult struct {
	Namespace string                  `json:"namespace"`
	Results   []NamespaceResultDetail `json:"results"`
}

type NamespaceResultDetail struct {
	Name       string      `json:"name"`
	Kind       string      `json:"kind"`
	PodResults []PodResult `json:"podResults"`
}

type PodResult struct {
	ID       string `json:"id"`
	Category string `json:"category"`
	Message  string `json:"message"`
	Severity string `json:"severity"`
	Success  bool   `json:"success"`
}
